src/js/actions/documents.js

```diff
@@ -1099,7 +1099,7 @@ define(function (require, exports) {
 
         return this.transfer(initActiveDocument);
     };
-    beforeStartup.reads = [locks.PS_DOC];
+    beforeStartup.reads = [locks.PS_DOC, locks.JS_APP];
```

Comment:
If action A transfers to an action B which holds lock X, then action A executes with lock X as well.